diff -ru a/include/shrinkwrap/istream.hpp b/include/shrinkwrap/istream.hpp
--- a/include/shrinkwrap/istream.hpp	2022-04-15 20:22:16.000000000 +0100
+++ b/include/shrinkwrap/istream.hpp	2025-04-08 19:22:50.955521297 +0100
@@ -8,7 +8,7 @@
 
 #include <streambuf>
 #include <memory>
-
+#include <cstdint>
 namespace shrinkwrap
 {
   namespace detail
@@ -73,4 +73,4 @@
   };
 }
 
-#endif //SHRINKWRAP_ISTREAM_HPP
\ No newline at end of file
+#endif //SHRINKWRAP_ISTREAM_HPP
diff -ru a/include/shrinkwrap/xz.hpp b/include/shrinkwrap/xz.hpp
--- a/include/shrinkwrap/xz.hpp	2022-04-15 20:22:16.000000000 +0100
+++ b/include/shrinkwrap/xz.hpp	2025-04-08 19:23:27.367113229 +0100
@@ -9,6 +9,7 @@
 #include <assert.h>
 #include <iostream>
 #include <limits>
+#include <cstdint>
 #include <cstring>
 
 namespace shrinkwrap
@@ -588,4 +589,4 @@
   }
 }
 
-#endif //SHRINKWRAP_XZ_HPP
\ No newline at end of file
+#endif //SHRINKWRAP_XZ_HPP
